sharedbuffers/mapped_struct.py

```diff
@@ -575,22 +627,28 @@ def unpack_from(cls, buf, offs, idmap = None):
         else:
             pbuf = buf
         try:
-            fs_type = pbuf[offs]
-            if fs_type == 'm' or fs_type == 'M':
+            if not cython.compiled and python3:
+                fs_type = bytes([pbuf[offs]])
+            else:
+                fs_type = pbuf[offs]
```

Comment:
This is unnecessary. `pbuf` is an `uchar*` which means `pbuf[offs]` will be an `uchar` regardless of python version.

If this is catering to pure python runs, remember we don't need pure python support. We should drop it and avoid the complications it brings, which means you should revert these to the cython-native implementation.

We should probably switch all `if cython.compiled` into `assert cython.compiled` (in another PR) because we should really drop pure python support. It's way too hard to maintain and completely unnecessary.

Pure python support was meant to allow running natively in PyPy, but PyPy's cpyext improved to the point where it can run cythonized code quite robustly, so we can depend on cython and simplify all those pesky `if cython.compiled` cases.

So... for now, just forget about pure python tests. Worry about making it run in cython.